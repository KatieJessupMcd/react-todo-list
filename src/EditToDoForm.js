import React, { Component } from 'react';

class EditToDoForm extends Component {
  constructor(props) {
    super(props);
    this.state = { task: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.updateTodo(this.state.task, this.props.currentTaskId);
    this.setState({ task: '' });
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="todo">Edit:</label>
        <input
          name="task"
          placeholder={this.props.currentTaskName}
          value={this.state.task}
          onChange={this.handleChange}
        />
        <button type="submit">Update</button>
        <button type="submit">Cancel</button>
      </form>
    );
  }
}

export default EditToDoForm;
