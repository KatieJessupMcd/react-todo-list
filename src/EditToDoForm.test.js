import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditToDoForm from './EditToDoForm';

// smoke test
it('renders without crashing', function() {
  shallow(<EditToDoForm />);
});

// snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<EditToDoForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
