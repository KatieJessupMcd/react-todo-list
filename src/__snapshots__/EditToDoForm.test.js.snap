// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="todo"
  >
    Edit:
  </label>
  <input
    name="task"
    onChange={[Function]}
    value=""
  />
  <button
    type="submit"
  >
    Update
  </button>
  <button
    type="submit"
  >
    Cancel
  </button>
</form>
`;
